framework/src/main/java/org/checkerframework/common/wholeprograminference/WholeProgramInferenceScenes.java

```diff
@@ -332,17 +381,30 @@ private String getEnclosingClassName(LocalVariableNode localVariableNode) {
      * @param executableElement the ExecutableElement
      * @return the "flatname" of the class enclosing {@code executableElement}
      */
-    private String getEnclosingClassName(ExecutableElement executableElement) {
+    @SuppressWarnings("signature") // https://tinyurl.com/cfissue/3094
+    private @BinaryName String getEnclosingClassName(ExecutableElement executableElement) {
         return ((MethodSymbol) executableElement).enclClass().flatName().toString();
     }
 
     /**
-     * Returns the "flatname" of the class enclosing {@code variableElement}
+     * Updates the metadata stored in AClassWrapper for the given class.
      *
-     * @param variableElement the VariableElement
-     * @return the "flatname" of the class enclosing {@code variableElement}
+     * @param classSymbol the class for which to update metadata
+     * @param aClassWrapper the class representation in which the metadata is to be updated
      */
-    private String getEnclosingClassName(VariableElement variableElement) {
-        return ((VarSymbol) variableElement).enclClass().flatName().toString();
+    private void updateClassMetadata(ClassSymbol classSymbol, AClassWrapper aClassWrapper) {
```

Comment:
The arguments to this method feel out of order to me.  The `AClassWrapper` feels more like the "receiver" and is modified, so I would put it first.  That's also the order used by the overview section of the Javadoc.